repowiz-application/src/main/java/life/qbic/repowiz/application/cli/RepowizTool.java

```diff
@@ -23,60 +22,115 @@
 
 /**
  * Implementation of RepoWiz. Its command-line arguments are contained in instances of {@link RepowizCommand}.
+ *
+ * This class should be used to access RepoWiz and start the tool
+ *
+ * @since: 1.0.0
+ * @author: Jennifer Bödker
  */
-public class RepowizTool{
+public class RepowizTool {
 
     private static final Logger LOG = LogManager.getLogger(RepowizTool.class);
 
     private SubmissionController controller;
     private CommandlineView commandlineView = new RepoWizView();
 
-
-    public RepowizTool(String projectID, String config){
+    /**
+     * Sets up the infrastructure of RepoWiz by building the connection to the local database, assembling the controller class and change listener
+     *
+     * @param projectID of the project that needs to be prepared for an uploaded
+     * @param config    with data to connect to the local database
+     */
+    public RepowizTool(String projectID, String config) {
         // set up infrastructure classes
         OpenBisProjectSearcher searcher = setupLocalDatabaseConnection(config);
 
         List<String> repos = getImplementedRepositoriesAsList();
 
-        controller = new SubmissionController(commandlineView,projectID, searcher, new RepositoryLoaderJava(),repos);
+        controller = new SubmissionController(commandlineView, projectID, searcher, new RepositoryLoaderJava(), repos);
 
+        //set up ChangeListener to handle user answers
         UserAnswer answer = new UserAnswer();
         answer.addPropertyChangeListener(controller);
 
         commandlineView.setUserAnswer(answer);
     }
 
-    public RepowizTool(){
+    public RepowizTool() {
         //add spi provider here!
     }
 
-    public void executeFindRepository(){
+    /**
+     * Determines all repository plugins that can be accessed by RepoWiz
+     *
+     * @return a list of all repository plugins
+     */
+    private static List<String> getImplementedRepositoriesAsList() {
+        List<String> repos = new ArrayList<>();
+        String fileName = "services/RepositoryJsonFiles.txt";
+
+        try {
+            InputStream stream = SubmissionController.class.getClassLoader().getResourceAsStream(fileName);
+            assert stream != null;
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+            for (String line; (line = reader.readLine()) != null; ) {
+                repos.add(line);
+            }
```

Comment:
I like the following better:
```suggestion
            repos.addAll(reader.lines().collect(Collectors.toList()));
```
```suggestion
            repos = new ArrayList<>(reader.lines().collect(Collectors.toList()));
```
```suggestion
            repos = reader.lines().collect(Collectors.toList());
```
In the last case, you could even remove the default of an empty list.